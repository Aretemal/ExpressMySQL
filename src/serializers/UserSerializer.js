import Serializer from './Serializer.js';

class UserSerializer extends Serializer {
  type() {
    return 'User';
  }

  id() {
    return this.resource.userId;
  }

  collectionSerializer() {
    return {
      type: this.type(),
      id: this.id(),
      attributes: this.attributes(),
    };
  }

  attributes() {
    return {
      fullName: this.resource.fullName,
      lastName: this.resource.lastName,
      login: this.resource.login,
      status: this.resource.status,
      email: this.resource.email,
      ava: this.resource.ava,
    };
  }
}
export default UserSerializer;
