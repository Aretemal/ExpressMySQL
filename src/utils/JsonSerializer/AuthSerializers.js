import Serializers from './Serializers.js';

class AuthSerializers extends Serializers {
  registration(data, links) {
    return this.serialize(data, data.id, 'ObjectUser', links);
  }

  login(data, links) {
    const jsonObj = this.serialize(data.user, data.user.id, 'Token', links);
    jsonObj.data.token = data.token;
    return jsonObj;
  }
}
export default new AuthSerializers();
