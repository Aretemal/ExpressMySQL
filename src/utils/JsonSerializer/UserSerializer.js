import Serializer from './Serializer.js';

class UserSerializer extends Serializer {
  getOne(data, links) {
    return this.serializer(data, data.id, 'ObjectUser', links);
  }

  getAllUsers(data, links) {
    return this.serializer(data, data.userAuth.id, 'Array Users', links);
  }

  getStatus(data, links) {
    return this.serializer(data, data.id, 'Status', links);
  }
}
export default new UserSerializer();
