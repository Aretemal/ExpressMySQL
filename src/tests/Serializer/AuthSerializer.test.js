import { matchersWithOptions } from 'jest-json-schema';
import AuthSerializers from '../../serializers/AuthSerializer.js';
import UserSerializer from '../../serializers/UserSerializer.js';
import schema from './schema.json';

expect.extend(matchersWithOptions({
  verbose: true,
}));

describe('AuthSerializer :', () => {
  describe('Registration', () => {
    test('should return object with data about user', () => {
      const user = {
        userId: 1,
        login: 'Artem',
        firstName: 'aaaa',
        lastName: 'bbbb',
        email: '111@11.1',
        status: '1',
        ava: null,
      };

      const serializer = new UserSerializer(user);
      const data = serializer.serialize();

      expect(data).toMatchSchema(schema);
      expect(data).toMatchSnapshot();
    });
  });

  describe('Login', () => {
    test('should return object and token', () => {
      const token = 'TOKEN';

      const serializer = new AuthSerializers(token);
      const data = serializer.serialize();

      expect(data).toMatchSchema(schema);
      expect(data).toMatchSnapshot();
    });
  });
});
