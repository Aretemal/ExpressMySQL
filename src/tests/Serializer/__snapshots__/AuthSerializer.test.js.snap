// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthSerializer : Login should return object and token 1`] = `
{
  "data": [
    {
      "attributes": {
        "token": undefined,
      },
      "id": "0",
      "type": "Token",
    },
  ],
  "links": {
    "self": "1",
  },
}
`;

exports[`AuthSerializer : Registration should return object with data about user 1`] = `
{
  "data": [
    {
      "attributes": {
        "ava": null,
        "email": "111@11.1",
        "firstName": "aaaa",
        "lastName": "bbbb",
        "login": "Artem",
        "status": "1",
      },
      "id": "1",
      "type": "User",
    },
  ],
  "links": {
    "self": "1",
  },
}
`;
